src/go/cmd/token-vendor/api/v1/v1.go

```diff
@@ -255,6 +255,45 @@ func (h *HandlerContext) tokenOAuth2Handler(w http.ResponseWriter, r *http.Reque
 	w.Write(tokenBytes)
 }
 
+// Handle requests to verify if a given JWT is valid
+//
+// This handler provides a backend for common proxy servers forward authentication.
+// I.e. it allows robots to sign a JWT and present it to nginx for authentication.
+//
+// Method: GET
+// Headers:
+// - Authorization: Bearer ...
+func (h *HandlerContext) verifyJWTHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		api.ErrResponse(w, http.StatusMethodNotAllowed,
+			fmt.Sprintf("method %s not allowed, only %s", r.Method, http.MethodGet))
+		return
+	}
+
+	authHeader, ok := r.Header["Authorization"]
+	if !ok {
+		api.ErrResponse(w, http.StatusBadRequest,
+			"request did not provide Authorization header")
+		return
+	}
+
+	if len(authHeader) != 1 {
+		api.ErrResponse(w, http.StatusBadRequest,
+			fmt.Sprintf("%q auth headers provided. Only 1 allowed", len(authHeader)))
+		return
+	}
+
+	// Be slightly permissive here. Allow both forms
+	// Authorization: Bearer ...
+	// Authorization: ...
+	jwtString := strings.TrimPrefix(authHeader[0], "Bearer ")
+
+	if _, err := h.tv.ValidateJWT(r.Context(), jwtString); err != nil {
+		api.ErrResponse(w, http.StatusForbidden, "JWT not valid")
```

Comment:
slog.Error()?

Reply:
This happens on every validation error, including expired. So at most I'd say Warn.

Might be interesting to log some internal ones at a higher level, I'll take a look.

Reply:
I'm not so fussed about the error, and this is indeed pretty noisy already from other handlers. An alternative would be able to generate metrics for the different error codes. My main concern is about having a way to debug unexpected failures.